conf/database.properties

```
@@ -21,3 +21,9 @@ scalar.db.password=cassandra
 # Either EXTRA_READ or EXTRA_WRITE can be specified. EXTRA_READ is used by default.
 # If SNAPSHOT is specified in the property "scalar.db.consensus_commit.isolation_level", this is ignored.
 #scalar.db.consensus_commit.serializable_strategy=
+
+# Debug mode usable for Consensus Commit.
+# If set to "true", Get and Scan operations results will contain transaction metadata. To see the transaction metadata columns details for a given table, you can use the `DistributedTransactionAdmin.getTableMetadata()`
+# method which will return the table metadata augmented with the transaction metadata columns. Using the debug mode can be useful to investigate transaction related issues.
+# The default is false.
+#scalar.db.consensus_commit.debug=false
```

Is it more consistent to name it like this for boolean configurations?
```suggestion
#scalar.db.consensus_commit.debug.enabled=false
```

This is just a comment, but `debug` might sound too broad unless we are planning to add more debug features.
I don't come up with a good naming, but we could make it more specific like `consensus_commit.filter_metadata.enabled` or something.
(in such a case, true by default, though)

---

Ok, I agree, updating the name to something more specific is a good idea I think. . I will add the `enabled` as well too, thank you for pointing it out. 
`consensus_commit.filter_metadata.enabled` -> Personally I would understand that metadata are kept in this case, not removed. Using `filter_out` might get the point across more clearly.
What do you think about one of the following?
- `consensus_commit.filter_out_metadata_columns.enabled`
- `consensus_commit.omit_metadata_columns.enabled`

---

I think `filter` itself also means removing something as below, so `out` might not be necessary.
https://www.oxfordlearnersdictionaries.com/definition/english/filter_2

`filter_metadata_columns.enabled` or making it more nouny like `metadata_columns_filtering.enabled` ?

---

Ok, both seem good but for consistency `metadata_columns_filtering.enabled` is probably better.

---

IMHO, as we filter the metadata columns by default, if we use the `enabled` wording, I feel `include metadata` or `show metadata` or something like that is more intuitive than `filter metadata`. What do you think? @feeblefakie @Torch3333

---

That's a good point to discuss. I don't have a strong preference for that.
Is it common to use `false` by default for this kind of boolean configuration?

As far as I can see in the Spark doc, both `true` and `false` are set by default.
(e.g., `spark.network.crypto.enabled` (false by default), `spark.ui.xContentTypeOptions.enabled` (true by default))
https://spark.apache.org/docs/latest/security.html

---

`include metadata` is a good option I think. It feels more clear-cut than `filter`

---

> As far as I can see in the Spark doc, both true and false are set by default.

@feeblefakie Ah okay, so maybe it doesn't depend on the default value. `Enable metadata filtering` sounds to me that we should originally return the metadata, but we don't when metadata filtering is enabled. However, as we hide the metadata from users so that users can access transaction tables transparently, I think the correct is that we should not return the metadata basically, but we do when enabling including (or showing) metadata is enabled.

---

@brfrn169 @Torch3333 I think it makes sense, so something like `include_metadata` also sounds good to me. (show sounds a bit weird to me because it's up to users if they show the metadata or not. but it makes sense if it is SQL command-line tool option)

Ideally, it can be a nouny form, but `metadata_inclusion` sounds a bit unusual?

---

I'm okay with `metadata_inclusion`. But I'm not sure it sounds unusual. What do you think? @Torch3333

---

`metadata_inclusion` sounds indeed a bit unusual. Even though `include_metadata` may not be consistent with the rest of the configuration naming, I think using terms developers are more familiar with is more important.

---

@Torch3333 Okay, sure. Let's use `include_metadata` then.

---

To sum up, the full configuration name will be `scalar.db.consensus_commit.include_metadata.enabled`